unittests/test_config.py

```diff
@@ -12,10 +12,34 @@
 
 
 
+@pytest.fixture(params=['full', 'parts'])
+def site_config(request):
+    # `unittests/resources/config/settings.py` should be equivalent to loading
+    # the `unittests/resources/config/settings-part*.py` files
+    if request.param == 'full':
+        return config.load_config('unittests/resources/config/settings.py')
+    else:
+        return config.load_config(
+            'unittests/resources/config/settings-part1.py',
+            'unittests/resources/config/settings-part2.py',
+            'unittests/resources/config/settings-part3.py'
+        )
+
+
 def test_load_config_python():
     config.load_config('reframe/core/settings.py')
 
 
+def test_load_multiple_configs():
+    site1 = config.load_config('unittests/resources/config/settings.py')
+    site2 = config.load_config(
+        'unittests/resources/config/settings-part1.py',
+        'unittests/resources/config/settings-part2.py',
+        'unittests/resources/config/settings-part3.py'
+    )
```

Comment:
I would probably generate those on the fly through a pytest fixture if we also need to reuse them. It's too much hassle to keep four (!) almost the same configuration files. You load as raw json the `resources/settings.py`, change some stuff and then dump it again with a separate name.